test/functions.jl

```diff
@@ -194,6 +194,62 @@ using Statistics
         @test length(ndv) == 0
     end
 
+    @testset "map, map!" begin
+        nda = NamedDimsArray([11 12; 21 22], (:x, :y))
+
+        @test names(map(+, nda, nda, nda)) == (:x, :y)
+        @test names(map(+, nda, parent(nda), nda)) == (:x, :y)
+        @test names(map(+, parent(nda), nda)) == (:x, :y)
+
+        # this method only called based on first two arguments:
+        @test names(map(+, parent(nda), parent(nda), nda)) == (:_, :_)
+
+        # one-arg forms work without adding anything, @which map(sqrt, nda) # Base
+        @test names(map(sqrt, nda)) == (:x, :y)
+        @test foreach(sqrt, nda) === nothing
+
+        # map! may return a different wrapper of the same data, like sum!
+        semi = NamedDimsArray(rand(2,2), (:x, :_))
+        @test names(map!(sqrt, semi, nda)) == (:x, :y)
+        @test names(map!(sqrt, rand(2,2), nda)) == (:x, :y)
+
+        zed = similar(nda, Float64)
+        @test map!(sqrt, zed, nda) == sqrt.(nda)
+        @test zed[1,1] == sqrt(nda[1,1])
+
+        # mismatching names
+        @test_throws DimensionMismatch map(+, nda, transpose(nda))
+        @test_throws DimensionMismatch map(+, nda, parent(nda), nda, transpose(nda))
+        @test_throws DimensionMismatch map!(sqrt, semi, transpose(nda))
+
+        @test foreach(+, semi, nda) === nothing
+        @test_throws DimensionMismatch foreach(+, semi, transpose(nda))
+    end
+
+    @testset "collect(generator)" begin
+        nda = NamedDimsArray([11 12; 21 22], (:x, :y))
+        ndv = NamedDimsArray([10, 20, 30], (:z,))
+
+        @test names([sqrt(x) for x in nda]) == (:x, :y)
+
+        @test names([x^i for (i,x) in enumerate(ndv)]) == (:z,)
+        @test names([x^i for (i,x) in enumerate(nda)]) == (:x, :y)
+
+        # Iterators.product -- has all names
+        @test names([x+y for x in nda, y in ndv]) == (:x, :y, :z)
```

Comment:
woah, that is nice

Reply:
I made this work for 2 now, I mean if only the second factor is named.